addons/point_of_sale/models/pos_config.py

```diff
@@ -26,10 +26,23 @@ def _default_invoice_journal(self):
         return self.env['account.journal'].search([('type', '=', 'sale'), ('company_id', '=', self.env.company.id)], limit=1)
 
     def _default_payment_methods(self):
-        return self.env['pos.payment.method'].search([('split_transactions', '=', False), ('company_id', '=', self.env.company.id)])
+        """ Should only default to payment methods that are compatible to this config's company and currency.
+        """
+        return self.env['pos.payment.method'].search([
+            ('split_transactions', '=', False),
+            ('company_id', 'in', (False, self.env.company.id)),
```

Comment:
in false is not needed, even if it is not enforced, each payment method should have a company_id